lib/model/model.go

```diff
@@ -1338,6 +1339,15 @@ func (m *model) ccHandleFolders(folders []protocol.Folder, deviceCfg config.Devi
 	}
 
 	indexSenders.removeAllExcept(seenFolders)
+	// All current pending folders were touched above, so discard any with older timestamps
+	if expired, err := m.db.RemovePendingFoldersBeforeTime(deviceID, handleTime); err != nil {
+		l.Infof("Could not clean up pending folder entries: %v", err)
+	} else if expired > 0 {
+		m.evLogger.Log(events.ClusterPendingChanged, map[string]interface{}{
+			"device": deviceID.String(),
+			"count":  expired,
+		})
```

Comment:
I think it would be more useful if it contained what was added/removed etc, so that receivers of the event would not even need to go to the other endpoints to reconcile the current state.

Reply:
See above, I think trying to wrap everything into the event is duplicated effort. The event should just be regarded as a trigger to refresh the info. Handling the changes individually is also prone to get the local (GUI) cache out of sync with the endpoint data. 

Having a full-fledged FolderRejected event in addition is due to legacy and keeping compatibility. It's good to include the additional details there so it's usable immediately, not strictly requiring a refresh.

Reply:
I think we keep old events as they were, and new event should contain everything the consumer needs (in order to avoid hitting the endpoint), meaning list of additions removals etc.

I think the epiphany is that someday the ui will be completely event sourced, and the endpoints will only be called once on page load to bootstrap.

Reply:
Regarding the GUI code, I read that deleting properties from a JS object is bad for browser performance. So on that end, should it still do a filter-and-replace dance or is it okay to simply `delete(pendingFolders[folderID], deviceID)`?

Reply:
Idk, I don't think we microoptimise our js code.

Reply:
Addressed in latest commits.